t5/t5.py

```diff
@@ -108,30 +111,67 @@ def __call__(
         mask: Optional[mx.array],
         cache: Optional[Tuple[mx.array, mx.array]] = None,
     ) -> [mx.array, Tuple[mx.array, mx.array]]:
-        queries = self.query_proj(queries)
-        keys = self.key_proj(keys)
-        values = self.value_proj(values)
+        queries = self.q(queries)
+        keys = self.k(keys)
+        values = self.v(values)
 
         num_heads = self.num_heads
         B, L, _ = queries.shape
         _, S, _ = keys.shape
         queries = queries.reshape(B, L, num_heads, -1).transpose(0, 2, 1, 3)
-        keys = keys.reshape(B, S, num_heads, -1).transpose(0, 2, 3, 1)
+        keys = keys.reshape(B, S, num_heads, -1).transpose(0, 2, 1, 3)
         values = values.reshape(B, S, num_heads, -1).transpose(0, 2, 1, 3)
 
         if cache is not None:
             key_cache, value_cache = cache
-            keys = mx.concatenate([key_cache, keys], axis=3)
+            keys = mx.concatenate([key_cache, keys], axis=2)
             values = mx.concatenate([value_cache, values], axis=2)
 
-        # Dimensions are [batch x num heads x sequence x hidden dim]
-        scores = queries @ keys
-        if mask is not None:
-            scores = scores + mask.astype(scores.dtype)
+        # scale = math.sqrt(1 / queries.shape[-1])
+        # output = mx.fast.scaled_dot_product_attention(
+        #     queries, keys, values, scale=scale, mask=mask
+        # )
+        # output = output.transpose(0, 2, 1, 3).reshape(B, L, -1)
```

Comment:
Why not use the above and instead write the computation explicitly?

Reply:
When using `fast.scaled_dot_product_attention` the output of the model becomes wrong, I left the code there so someone familiar with the fast version can check what went wrong.